spm/spm_internal.h

```diff
@@ -18,91 +18,85 @@
+#define SPM_COMPLETION_SEM_INITIAL_COUNT (0UL) /* Initial number of available tokens for a completion semaphore. */
 
-#define SPM_COMPLETION_SEM_MAX_COUNT (1UL) /**< Maximum number of available tokens for a completion semaphore.*/
-#define SPM_COMPLETION_SEM_INITIAL_COUNT (0UL) /**< Initial number of available tokens for a completion semaphore.*/
+#define PSA_MMIO_PERM_READ_ONLY   (0x000000001)
+#define PSA_MMIO_PERM_READ_WRITE  (0x000000003)
 
-typedef struct spm_partition spm_partition_t;
-typedef struct spm_ipc_channel spm_ipc_channel_t;
+#define PSA_RESERVED_ERROR_MIN (INT32_MIN + 1)
+#define PSA_RESERVED_ERROR_MAX (INT32_MIN + 127)
 
-/*
- * Enumeration for the possible channel processing states.
- */
-typedef enum partition_state {
-    CHANNEL_STATE_INVALID = 0,
-    CHANNEL_STATE_CONNECTING = 1,
-    CHANNEL_STATE_IDLE = 2,
-    CHANNEL_STATE_PENDING = 3,
-    CHANNEL_STATE_ACTIVE = 4
-} ChannelState;
+#define SPM_CHANNEL_STATE_INVALID          (0x01)
+#define SPM_CHANNEL_STATE_CONNECTING       (0x02)
+#define SPM_CHANNEL_STATE_IDLE             (0x03)
+#define SPM_CHANNEL_STATE_PENDING          (0x04)
+#define SPM_CHANNEL_STATE_ACTIVE           (0x05)
 
-/*
- * Structure containing the SPM internal data.
- */
-typedef struct spm_db {
-    spm_partition_t *partitions; /* Array of all the Secure Partitions in the system.*/
-    uint32_t partition_count; /* Number of Secure Partitions in the system.*/
-    psa_handle_manager_t channels_handle_mgr;
-    psa_handle_manager_t messages_handle_mgr;
-    osMemoryPoolId_t channel_mem_pool; /* Channel memory pool identifier.*/
-    osMemoryPoolId_t active_messages_mem_pool; /* Channel memory pool identifier.*/
-} spm_db_t;
+
+#ifndef TRUE
+#define TRUE (1)
+#endif
+
+#ifndef FALSE
+#define FALSE (0)
+#endif
+
+struct spm_partition;
+struct spm_ipc_channel;
+
+typedef union spm_iovec {
+    psa_invec_t in;
+    psa_outvec_t out;
+} spm_iovec_t;
 
 /*
- * Structure containing the currently active message for a Root of Trust Service.
+ * IRQ signal mapper definition.
+ * The function will not return on invalid signal.
  */
-typedef struct spm_active_msg {
-    spm_ipc_channel_t *channel; /* Pointer to the channel delivering this message.*/
-    psa_invec_t in_vec[PSA_MAX_INVEC_LEN]; /* IOvecs sent.*/
-    psa_outvec_t out_vec[PSA_MAX_OUTVEC_LEN]; /* IOvecs for response.*/
-    psa_error_t rc; /* Return code to be filled by the Root of Trust Service.*/
-    SpmMsgType type; /* The message type, one of ::spm_msg_type.*/
-} spm_active_msg_t;
+typedef IRQn_Type (*spm_signal_to_irq_mapper_t)(uint32_t);
 
+// spm_pending_call_msg struct below is packed since in a dual processor solution
+// it is used in both processors
 /*
  * Structure containing data sent from NSPE for ROT_SRV call.
  */
-typedef struct spm_pending_call_msg {
-    psa_handle_t channel; /* Handle pointing to the channel */
+typedef __PACKED_STRUCT spm_pending_call_msg {
```

Comment:
Note, we provide these in mbed_toolchain, same for aligned and others

Reply:
We try to minimize the dependency on mbed tools inside SPM as we plan to contribute this implementation to TF-M.
It also helps us to start prototyping with platforms which are not yet supported by mbed-os